src/core/coap/secure_coap_server.cpp

```diff
@@ -117,7 +117,8 @@ void SecureServer::Receive(Message &aMessage, const Ip6::MessageInfo &aMessageIn
     {
         mPeerAddress.SetPeerAddr(aMessageInfo.GetPeerAddr());
         mPeerAddress.SetPeerPort(aMessageInfo.GetPeerPort());
-
+        mPeerAddress.SetSockAddr(aMessageInfo.GetSockAddr());
```

Comment:
Do we also need to enable macro "OPENTHREAD_ENABLE_BORDER_AGENT" here? If we do not enable border agent, maybe we do not need to set source address and port.